### PR TITLE
Reservations RP 2018-06-01: Add CosmosDb in reserved resource enum type. Add name property in PatchProperties

### PR DESCRIPTION
**Pull request by luluRagdoll**
## Description

Add CosmosDb in reserved resource enum type.
Add name property in PatchProperties

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by luluRagdoll** [comment]
@cormacpayne Hi Cormac,
Since the .NET SDK Microsoft.Azure.Management.Reservations 1.9.0-preview version is not published, I added in the localfeed to get unblocked. 
The change is to add CosmosDb as a new type in the reserved resource enum type.
Add name property in the Patch properties.

Could you help take a look?Thx

**Comment by maddieclayton** [comment]
@luluRagdoll please link your cmdlet design review from here: https://github.com/Azure/azure-powershell-cmdlet-review-pr

**Comment by luluRagdoll** [comment]
@maddieclayton Sure. I will work on that and link it. :)

**Comment by luluRagdoll** [comment]
@maddieclayton This is the design review link: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/154 Please feel free to let me know if you have any questions. Thanks

File: src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservation.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Get specific `Reservation` details.
 The credentials, account, tenant, and subscription used for communication with Azure.
 
 ```yaml
-Type: Microsoft.Azure.Commands.Common.Authentication.Abstractions.IAzureContextContainer
+Type: IAzureContextContainer
```
Comment (MiYanni) [inline src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservation.md]: You need to regenerate all the help files using the `UseFullTypeName` flag. Details here: https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module

File: src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/PatchReservation.cs [hunk 2]
```diff
@@ -49,6 +49,10 @@ public class PatchReservation : AzureReservationsCmdletBase
         [ValidateNotNull]
         public PSReservation Reservation { get; set; }
 
+        [Parameter(Mandatory = false)]
+        [ValidateNotNull]
+        public string Name { get; set; }
```
Comment (MiYanni) [inline src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/PatchReservation.cs]: I don't see updated help documentation for the `Name` parameter. Also, please provide an example in the help using this parameter.

File: src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/GetCatalog.cs [hunk 3]
```diff
@@ -17,7 +17,7 @@ public class GetCatalog : AzureReservationsCmdletBase
         public Guid SubscriptionId { get; set; }
 
         [Parameter(Mandatory = true)]
-        [PSArgumentCompleter("VirtualMachines", "SqlDatabases", "SuseLinux")]
+        [PSArgumentCompleter("VirtualMachines", "SqlDatabases", "SuseLinux", "CosmosDb")]
```
Comment (MiYanni) [inline src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/GetCatalog.cs]: I don't see updated help documentation with a new valid value. Looks like help may have been generated prior to these changes.

**Comment by maddieclayton** [comment]
@MiYanni this will need a new branch for preview release.

**Comment by MiYanni** [comment]
@maddieclayton I'll create a branch when this is ready for merging. As it stands, the design review hasn't been accepted yet.

File: src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservation.md (hunk 1 above)
Comment (MiYanni) [reply src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservation.md]: 1. In your changelog, `src\ResourceManager\Reservations\ChangeLog.md`, you add an entry under the `Current Release` header.
2. Sounds good.

File: src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/GetCatalog.cs (hunk 3 above)
Comment (luluRagdoll) [reply src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/GetCatalog.cs]: Checked in the help document "Get-AzureRmReservationCatalog.md". There are examples. But the document doesn't list all the examples for each reserved resource type. Should have to examples for all the RI types?

File: src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservation.md (hunk 1 above)
Comment (luluRagdoll) [reply src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservation.md]: Thanks. DONE.

File: src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/GetCatalog.cs (hunk 3 above)
Comment (MiYanni) [reply src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/GetCatalog.cs]: My fault. I thought that completer values are listed in the reference documentation. You do not need examples for each ReservedResourceType.

File: src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservation.md (hunk 1 above)
Comment (MiYanni) [reply src/ResourceManager/Reservations/Commands.Reservations/help/Get-AzureRmReservation.md]: You didn't regenerate the help documentation using the `UseFullTypeName` flag.

**Comment by luluRagdoll** [comment]
I see. Looks like I should not use the command that our team has in the doc. Will follow https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module to get the full name.

File: src/ResourceManager/Reservations/Commands.Reservations/ChangeLog.md [hunk 4]
```diff
@@ -19,6 +19,10 @@
         - Additional information about change #1
 -->
 ## Current Release
+## Version 0.1.7
```
Comment (MiYanni) [inline src/ResourceManager/Reservations/Commands.Reservations/ChangeLog.md]: Please remove the `## Version 0.1.7` tag. Just keep the entries under the `## Current Release` tag.

Comment (luluRagdoll) [reply src/ResourceManager/Reservations/Commands.Reservations/ChangeLog.md]: Sure. Done.

**Comment by MiYanni** [comment]
@luluRagdoll 
I've accepted the PR. However, the build failed. Please look into the details and fix the issues.

**Comment by luluRagdoll** [comment]
Sure. Taking a look now.

**Comment by MiYanni** [comment]
@luluRagdoll Any update? It looks like 1 test is failing, `TestGetCatalog`.

**Comment by luluRagdoll** [comment]
Hi Michael,
I have root cause the issue. Will update this today. 
Thanks
Yulu

File: src/ResourceManager/Reservations/Commands.Reservations/help/Update-AzureRmReservation.md [hunk 5]
```diff
@@ -107,6 +107,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -Name
+{{Fill Name Description}}
```
Comment (MiYanni) [inline src/ResourceManager/Reservations/Commands.Reservations/help/Update-AzureRmReservation.md]: This is missing a description for this parameter.

**Comment by MiYanni** [comment]
@luluRagdoll Build is passing, but just a minor change to the help documentation is needed. If you make this change, I can release the module to the test gallery today.

**Comment by MiYanni** [comment]
demand: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/1111/
ps-sign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/1369/
psnetsign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/212/

**Pull request merged**